clis/teliod/src/qnap.rs

```diff
@@ -0,0 +1,398 @@
+use std::{
+    fs, io,
+    path::PathBuf,
+    process::{Command, Stdio},
+    time::Duration,
+};
+
+use const_format::concatcp;
+use rust_cgi::{http::Method, text_response, Request, Response};
+use tracing::level_filters::LevelFilter;
+use uuid::Uuid;
+
+use crate::{
+    command_listener::CommandResponse,
+    config::{InterfaceConfig, MqttConfig, TeliodDaemonConfig},
+    ClientCmd, DaemonSocket, TIMEOUT_SEC,
+};
+
+const TELIOD_TMP_DIR: &str = "/tmp/nordsecuritymeshnet";
+const PID_FILE: &str = concatcp!(TELIOD_TMP_DIR, "/teliod.pid");
+const QPKG_DIR: &str = "/share/CACHEDEV1_DATA/.qpkg/NordSecurityMeshnet";
+const TELIOD_BIN: &str = concatcp!(QPKG_DIR, "/teliod");
+const MESHNET_LOG: &str = concatcp!(QPKG_DIR, "/meshnet.log");
+const TELIOD_LOG: &str = "/var/log/teliod.log";
+
+#[cfg(not(test))]
+const TELIOD_CFG: &str = concatcp!(QPKG_DIR, "/teliod.cfg");
+#[cfg(test)]
+use tests::TELIOD_CFG;
+
+pub(crate) fn handle_request(request: Request) -> Response {
+    match (request.method(), request.uri().query()) {
+        (&Method::POST, _) => start_daemon(),
+        (&Method::DELETE, _) => stop_daemon(),
+        (&Method::PATCH, _) => {
+            let body = match String::from_utf8(request.into_body()) {
+                Ok(body) => body,
+                Err(_) => return text_response(400, "Invalid UTF-8 in request body."),
+            };
+            update_config(&body)
+        }
+        (&Method::GET, Some("info=get-status")) => get_status(),
+        (&Method::GET, Some("info=get-teliod-logs")) => get_teliod_logs(),
+        (&Method::GET, Some("info=get-meshnet-logs")) => get_meshnet_logs(),
+        (_, _) => text_response(400, "Invalid request."),
+    }
+}
+
+fn is_teliod_running() -> bool {
+    fs::read_to_string(PID_FILE)
+        .and_then(|pid| fs::metadata(format!("/proc/{}", pid.trim())))
+        .is_ok()
+}
+
+fn teliod_socket_exists() -> bool {
+    DaemonSocket::get_ipc_socket_path()
+        .and_then(|path| path.metadata())
+        .is_ok()
+}
+
+fn kill_teliod_process() -> Result<(), io::Error> {
+    match fs::read_to_string(PID_FILE) {
+        Ok(pid) => {
+            let _ = Command::new("kill").arg("-9").arg(pid.trim()).status();
```

Comment:
added

Reply:
Referring to the above comment, this doesn't guarantee that the process being killed is effectively the telio daemon process right?

Reply:
Yeah, I'm thinking that maybe we shouldn't be calling `kill` at all - it might turn out it's possible somehow to trick our cgi handler to kill arbitrary processes on qnap.

Reply:
Removed `kill` process measure